pydantic_extra_types/types/country.py

```diff
@@ -0,0 +1,1185 @@
+"""
+Country definitions that are based on the ISO 3166 format
+Based on: https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes
+"""
+import re
+from enum import Enum
+from typing import Any, Dict, Optional, Tuple, Union
+
+from pydantic_core import PydanticCustomError, core_schema
+
+from pydantic import constr
+from pydantic._internal import _repr
+
+"""
```

Comment:
instead of all these different dicts with duplicated data, I would instead implement this as follows:

```py
from dataclasses import dataclass
from functools import lru_cache


@dataclass  # this is a standard dataclass
class CountryInfo:
    alpha2: str
    alpha3: str
    numeric_code: str
    short_name: str
    official_name: str


# U don't actually know if its worth making countries a function rather than a module
# level variable, best to check
@lru_cache()
def _countries() -> list[CountryInfo]:
    return [
        CountryInfo(
            'AF',
            'AFG',
            '004',
            'Afghanistan',
            'The Islamic Republic of Afghanistan',
        ),
        ...
    ]


@lru_cache()
def _index_by_alpha2() -> dict[str, CountryInfo]:
    return {country.alpha2: country for country in _countries()}

# other indexes
```

This should have much less duplication than the current logic.